packages/big-design/src/components/Tooltip/Tooltip.tsx

```diff
@@ -13,90 +14,80 @@ export interface TooltipProps extends React.HTMLAttributes<HTMLDivElement> {
   inline?: boolean;
 }
 
-interface State {
-  visible: boolean;
-}
+export const Tooltip: React.FC<TooltipProps> = typedMemo(
+  ({ children, inline = true, modifiers, trigger, ...props }) => {
+    const [isVisible, setIsVisible] = useState(false);
+    const [tooltipContainer, setTooltipContainer] = useState<HTMLDivElement | null>(null);
+
+    useEffect(() => {
+      const container = document.createElement('div');
 
-export class Tooltip extends React.PureComponent<TooltipProps, State> {
-  static defaultProps: Partial<TooltipProps> = {
-    placement: 'top',
-    inline: true,
-  };
+      document.body.appendChild(container);
+      setTooltipContainer(container);
+    }, []);
 
-  state = {
-    visible: false,
-  };
+    useEffect(() => {
+      return () => {
+        if (tooltipContainer) {
+          document.body.removeChild(tooltipContainer);
+        }
+      };
+    }, [tooltipContainer]);
```

Comment:
I don't understand the purpose of this. You set the tooltipContainer up top but then remove it?

Reply:
Yes - this useEffect just handles the unmount.